### PR TITLE
Add and use internal::FEPointEvaluation::is_fast_path_supported

### PR DESCRIPTION
**Pull request by peterrum**
references #12014 and (should fix simplex/wedge/pyramid)

### CODE REVIEW

**Review by kronbichler — COMMENTED** [review]
Should we use this one also for https://github.com/dealii/dealii/blob/master/include/deal.II/fe/fe_point_evaluation.h#L561

**Comment by peterrum** [comment]
@kronbichler But this is what I am using it for. Or am I missing something?

File: include/deal.II/fe/fe_point_evaluation.h [hunk 1]
```diff
@@ -558,7 +562,7 @@ FEPointEvaluation<n_components, dim, spacedim>::FEPointEvaluation(
   , fe(&fe)
 {
   if (mapping_q_generic != nullptr &&
-      internal::MatrixFreeFunctions::ShapeInfo<double>::is_supported(fe))
+      internal::FEPointEvaluation::is_fast_path_supported(fe))
```
Comment (kronbichler) [inline include/deal.II/fe/fe_point_evaluation.h:565]: Indeed, I must have missed this in front of all the other changes, my bad. :+1:

**Review by kronbichler — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]
Looks OK to me.

**Pull request merged**